pom.xml

```diff
@@ -35,52 +34,53 @@
 
     <modules>
         <module>bridge</module>
+        <module>bom</module>
         <module>distribution</module>
         <module>management</module>
-        <module>resource.address</module>
-        <module>resource.address.http</module>
-        <module>resource.address.httpx</module>
-        <module>resource.address.httpxdraft</module>
-        <module>resource.address.httpxe</module>
-        <module>resource.address.pipe</module>
-        <module>resource.address.rtmp</module>
-        <module>resource.address.sse</module>
-        <module>resource.address.ssl</module>
-        <module>resource.address.tcp</module>
-        <module>resource.address.udp</module>
-        <module>resource.address.ws</module>
-        <module>resource.address.wsdraft</module>
-        <module>resource.address.wse</module>
-        <module>resource.address.wsn</module>
-        <module>resource.address.wsr</module>
-        <module>resource.address.wsx</module>
-        <module>resource.address.wsxdraft</module>
+        <module>resource.address/spi</module>
+        <module>resource.address/http</module>
+        <module>resource.address/httpx</module>
+        <module>resource.address/httpxdraft</module>
+        <module>resource.address/httpxe</module>
+        <module>resource.address/pipe</module>
+        <module>resource.address/rtmp</module>
+        <module>resource.address/sse</module>
+        <module>resource.address/ssl</module>
+        <module>resource.address/tcp</module>
+        <module>resource.address/udp</module>
+        <module>resource.address/ws</module>
+        <module>resource.address/wsdraft</module>
+        <module>resource.address/wse</module>
+        <module>resource.address/wsn</module>
+        <module>resource.address/wsr</module>
+        <module>resource.address/wsx</module>
+        <module>resource.address/wsxdraft</module>
         <module>security</module>
         <module>server</module>
         <module>server.api</module>
         <module>server.demo</module>
         <module>server.spi</module>
-        <module>service</module>
-        <module>service.amqp</module>
-        <module>service.broadcast</module>
-        <module>service.echo</module>
-        <module>service.http.balancer</module>
-        <module>service.http.directory</module>
-        <module>service.http.proxy</module>
-        <module>service.proxy</module>
-        <module>service.update.check</module>
-        <module>service.update.check.management</module>
-        <module>transport</module>
-        <module>transport.bio</module>
-        <module>transport.http</module>
-        <module>transport.nio</module>
-        <module>transport.pipe</module>
-        <module>transport.sse</module>
-        <module>transport.ssl</module>
-        <module>transport.ws</module>
-        <module>transport.wseb</module>
-        <module>transport.wsn</module>
-        <module>transport.wsr</module>
+        <module>service/spi</module>
```

Comment:
I am ok with this way of organizing things under transport, service, resource.address. Other way is organize under protocols (for e.g: http, http/resource.address, http/transport, http/service.http.balancer, http/service.http.proxy etc). I wonder whether we thought about that one.